main.cc

```diff
@@ -1173,6 +1174,15 @@ sharded<locator::shared_token_metadata> token_metadata;
             auto stop_dsm = defer_verbose_shutdown("disk space monitor", [&disk_space_monitor_shard0] {
                 disk_space_monitor_shard0->stop().get();
             });
+            auto out_of_space_subscription = disk_space_monitor_shard0->subscribe(cfg->critical_disk_utilization_level, [&threhsold = cfg->critical_disk_utilization_level, &dsm = *disk_space_monitor_shard0] (auto threshold_reached) {
```

Comment:
nit: why `auto`? it wpuld be clearer to specify the real type here

Reply:
Why do you need a real type here? The variable is unused. It is there only to ensure the subscription is not automatically removed. So `auto` is actually more clear to me in this case.

Reply:
> Why do you need a real type here? The variable is unused. It is there only to ensure the subscription is not automatically removed. So `auto` is actually more clear to me in this case.

I meant in `auto threshold_reached`, but it's not critical.